### ISSUE TITLE
[MU4 Issue] Difficult to shut MS4 after using other programs (e.g. Firefox browser)

### ISSUE THREAD

**Issue by rgreen5** [musescore/MuseScore#15714]
**Describe the bug**

This problem has ocurred repeatedly over several months. It seems to happen when you leave MS4 open with a score in progress, then use another program such as Mozilla Firefox, then try to close MS4.

**To Reproduce**

The following _may_ work for you:

1. Open MS4
2. Create a treble staff score. 
3. Enter a few notes etc.
4. Open the browser and use it for perhaps ten minutes or more (the length of time may, or may not, be a factor)
5. Click back into MS4, then try to shut the program by clicking on the window cross icon.

RESULT: The program hangs. It may take some time to close. Meanwhile the rest of the system may slow to a crawl.

**Expected behavior**

MS4 should close right away. response of rest of system should not be affected.

**Platform information**

OS: Linux Mint 20.1, Arch.: x86_64, MuseScore version (64-bit): 4.0.0-230040304, revision: github-musescore-musescore-eab50c9

**Comment by rgreen5** [comment]
OS: Linux Mint 20.1, Arch.: x86_64, MuseScore version (64-bit): 4.0.2-230410504, revision: github-musescore-musescore-c1d5087

I'm still experiencing the same problem. Closing the program may take a minute to complete and meanwhile the system freezes.

**Comment by cbjeukendrup** [comment]
@DmitryArefiev Can you reproduce this?

**Comment by DmitryArefiev** [comment]
> @DmitryArefiev Can you reproduce this?

Nope. No hang on my side (LinuxMint 21 Cinnamon, 5.4.12, latest 4.0.2 build)

**Comment by MarcSabatella** [comment]
FWIW, I see something kind of similar on one of two Linux systems which are otherwise pretty identical (both are Debian containers within Chromebooks).  It started, I think, with 4.0.1 - or perhaps with an OS update around that same time.  When I close MuseScore, at least half the time, it leaves behind a zombie process, and not just for a minute or so - it stays running until I kill it.  The UI closes normally and there is no indication anything is still running except sometimes it uses enough CPU for me to notice.  And of course, since multiple scores open multiple instances, I sometimes find myself with a dozen of these zombie processes - that's when I'm likely to notice the problem.  I've gotten into the habit of just running "pkill mscore" periodically.

I know, this system configuration is unusual, and I haven't a clue what's actually triggering the issue.  Usually I'm starting MuseScore from a task bar icon so there is no console output to report.  When I've tried to catch it in the act by running from the command line, it closes normally.  I'll try some more and see if I can see any interesting console output that would provide a clue.

**Comment by laturetab** [comment]
I have a similar thing happening on Windows 11, running debug (or relwithdebinfo). It happens about half the time. 
The program crashes on exit. Always in the same place: in the destructor of NetworkManager.cpp around line 45. For me, m_reply is null so something else being destructed is wiping the program out on shutdown (sometimes).

**Comment by laturetab** [comment]
I just had this happen again.

Essentially, Qt5Network.dll is throwing an exception as MuseScore is shutting down. It always happens in the destructor of NetworkManager.cpp.  I included the Visual Studio message. This has happened consistently for about a month, so it's not something added recently that's causing the problem.

Exception thrown at 0x00007FF98162682D (Qt5Network.dll) in MuseScore4.exe: 0xC0000005: Access violation reading location 0x00007FF9C301E3C8.

**Comment by MarcSabatella** [comment]
I had almost started to think I wasn't going to be able to reproduce the problem starting the 4.0.2 AppImage from the command line as I went several days without seeing it, but it just happened.  Right now on my system, all MuseScore windows are closed, and I have the terminal window open that I started it from, but it's hanging there, and another terminal window shows the process still running.

Here is a text file with the console output from the point I tried quitting:

[console.txt](https://github.com/musescore/MuseScore/files/10798894/console.txt)

**Comment by Eism** [comment]
@laturetab 
Can you reproduce on the stable version? Because may be you have something incorrectly configured in your environment and this is a completely different error.

Regarding the bug - I can't reproduce on any platform.

**Comment by laturetab** [comment]
@Eism 
You have to run in Debug in Visual Studio and be patient. It may take 10-20 tries before it happens. However, there were some changes checked into master last week that may have fixed it. I picked up those changes yesterday and haven't seen the problem since. I will post here if it happens again. 

You may want to look at the last 80 or so commits to master to find it -- anything to do with multi threading and shutdown -- I noticed a couple of them last week. e.g. initializing COM single threaded

**Comment by laturetab** [comment]
@Eism,
The commit that fixed this for me was "#16472 - fixed crash on deinit" by @igorkorsukov on Feb 22.

**Comment by rgreen5** [comment]
OS: Linux Mint 20.1, Arch.: x86_64, MuseScore version (64-bit): 4.1.0-231791358, revision: github-musescore-musescore-dfc286f

Experienced this problem again with MS4.1 beta open (and in use) with Firefox also in use. MS4 is very slow to close.

**Comment by howdev** [comment]
Yes, I also agree, Musescore Studio 4.4.4 is also slow to close. but no force quit.
I thought it may be appimage, however, I also tried the old deb package 3.2, it is also slow to close. Therefore, it is the app itself is not responsive to closing. 
It is even worse when the app is pin to the dock.  The wait icon shows for 8 seconds, you need to wait that long to be able to start the app after each closing. clicking immediately to launch the app does nothing.
